### ISSUE TITLE
Customized Loss Function

### ISSUE THREAD

**Issue by maritum** [open-mmlab/mmdetection#10494]
**Describe the feature**

**Motivation**
I work on Masked R-CNN model, where cross entropy loss is used for mask head in default config.
In order to address class imbalance problem, I want to use Dice Loss for mask head. However, there is a tensor size mismatch, while calculating dice_loss. 

 For example, `mask_cross_entropy` is defaultly used for FCNMaskHead where only the mask corresponding to the ground truth class label is selected to compute its mask loss instead of all 35 masks. 

```
def mask_cross_entropy(pred,
                       target,
                       label,
                       reduction='mean',
                       avg_factor=None,
                       class_weight=None,
                       ignore_index=None,
                       **kwargs):


    assert ignore_index is None, 'BCE loss does not support ignore_index'
    # TODO: handle these two reserved arguments
    assert reduction == 'mean' and avg_factor is None
    num_rois = pred.size()[0]
    inds = torch.arange(0, num_rois, dtype=torch.long, device=pred.device)
    pred_slice = pred[inds, label].squeeze(1)
    return F.binary_cross_entropy_with_logits(
        pred_slice, target, weight=class_weight, reduction='mean')[None]
````

According to the source code, DiceLoss doesn't seem to have this key step. 

My question is, can these changes be taken into account in Dice_loss, or can you provide instructions on how to register a custom loss function. Thank you!
